tar/bsdtar.c

```diff
@@ -416,7 +417,9 @@ main(int argc, char **argv)
 			optq_push(bsdtar, "exclude", bsdtar->optarg);
 			break;
 		case 'f': /* multitar */
-			bsdtar->tapenames[bsdtar->ntapes++] = bsdtar->optarg;
+			tapename_cmdline = (char *)bsdtar->optarg;
```

Comment:
`clang -Weverything` is quite happy with it...?  Anyway, I split the de-consting into #267 and tweaked the intermediate commits.